functions/meme-getter/tests/memeGetter.test.js

```diff
@@ -0,0 +1,266 @@
+				ID: 56,
+				bottomText: '',
+				image: 'https://imgflip.com/s/meme/Philosoraptor.jpg',
+				name: 'Philosoraptor',
+				rank: 165,
+				tags: 'What if, dinosaur, Velociraptor, metaphysical inquiries',
+				topText: 'What if ...',
+			},
+			{
+				ID: 17,
+				bottomText: 'Everywhere',
+				image: 'https://imgflip.com/s/meme/X-Everywhere.jpg',
+				name: 'X, X Everywhere',
+				rank: 170,
+				tags: 'Toy story, Buzz lightyear, Woody',
+				topText: '___, ___',
+			},
+			{
+				ID: 20,
+				bottomText: "That'd be great",
+				image: 'https://imgflip.com/s/meme/That-Would-Be-Great.jpg',
+				name: 'That would be great',
+				rank: 180,
+				tags: 'Bill Lumbergh, Office Space',
+				topText: 'Yeah, If you could just ___',
+			},
+			{
+				ID: 18,
+				bottomText: '',
+				image: 'https://imgflip.com/s/meme/Good-Guy-Greg.jpg',
+				name: 'Good Guy Greg',
+				rank: 180,
+				tags: 'ggg',
+				topText: '      ',
+			},
+			{
+				ID: 258,
+				bottomText: 'Show Me What You Got',
+				image: 'http://i.imgur.com/6Ln3hp8.png',
+				name: 'Show Me What You Got',
+				rank: 188,
+				tags: 'Rick and Morty, Giant Heads, Dan, Justin, Adult Swim',
+				topText: '',
+			},
+			{
+				ID: 21,
+				bottomText: '',
+				image: 'https://imgflip.com/readImage?iid=21808196',
+				name: 'What if I told you',
+				rank: 200,
+				tags: 'Matrix, Morpheus, Lawrence Fishburne',
+				topText: 'What if I Told You',
+			},
+			{
+				ID: 55,
+				bottomText: "Ain't nobody got time for that",
+				image: 'https://imgflip.com/s/meme/Aint-Nobody-Got-Time-For-That.jpg',
+				name: "Ain't nobody got time for that",
+				rank: 205,
+				tags: 'sweet brown Kimberly Wilkins',
+				topText: '',
+			},
+			{
+				ID: 54,
+				bottomText: '',
+				image: 'https://imgflip.com/s/meme/Insanity-Wolf.jpg',
+				name: 'Insanity Wolf',
+				rank: 205,
+				tags: 'animal, insane, dog',
+				topText: '',
+			},
+		],
+		message: 'GET successful',
+		next: 'http://alpha-meme-maker.herokuapp.com/2',
+	};
+	const MOCK_FAILURE_RESPONSE = {
+		errorCode: 500,
+		errMessage: 'ValidationError: "page" must be a number',
+	};
+	beforeAll(() => {
+		event = data;
+	});
+	it('should callled the handler', async () => {
+		const memeSpy = jest
+			.spyOn(MemeService, 'getMemes')
+			.mockResolvedValue(MOCK_SUCCESS_RESPONSE);
+		const apiSucessSpy = jest
+			.spyOn(utils, 'apiSuccess')
+			.mockReturnValue(MOCK_SUCCESS_RESPONSE);
+		const callbackFn = () => {};
+		const result = await handler(event, {}, callbackFn);
+		expect(memeSpy).toBeCalledWith(event.pathParameters.page);
+		expect(result).toEqual(MOCK_SUCCESS_RESPONSE);
+		expect(apiSucessSpy).toBeCalledWith(callbackFn, MOCK_SUCCESS_RESPONSE);
+	});
+
+	it('should throw error when page is passed with anyother datatype except number', async () => {
+		event.pathParameters = '1';
+		jest.spyOn(utils, 'apiFailure').mockReturnValue(MOCK_FAILURE_RESPONSE);
+		const callbackFn = () => {};
+		const result = await handler(event, {}, callbackFn);
```

Comment:
can we assign the "result" variable at the top with "let" and keep updating it instead of decalring it with const eveyr time?

ref :- line 250, line 260

Reply:
you can do this for callbackFn too, assign const callbackFn = jest.fn(); at the top, since its same throughout both tests

Reply:
Done.